avm/src/vm/deploy.rs

```diff
@@ -21,40 +21,46 @@ struct AWSCredentials {
   secretAccessKey: String,
 }
 
+#[allow(non_snake_case)]
 #[derive(Deserialize, Debug, Serialize)]
 struct AWSConfig {
   credentials: AWSCredentials,
   region: String,
+  cloudAlias: String,
 }
 
 #[derive(Deserialize, Debug, Serialize)]
-struct Config {
-  aws: AWSConfig,
+struct DeployConfig {
+  aws: Vec<AWSConfig>,
 }
 
+#[allow(non_snake_case)]
 #[derive(Deserialize, Debug)]
 struct App {
   id: String,
   url: String,
   status: String,
+  cloudProvider: String,
+  cloudAlias: String,
 }
 
 const CONFIG_NAME: &str = ".alan/deploy.json";
 const CONFIG_SCHEMA: &str = "Please define a deploy config with the following schema: \n{
-  \"aws\": {
-      \"region\": \"string\",
-      \"credentials\": {
-        \"accessKeyId\": \"string\",
-        \"secretAccessKey\": \"string\",
-      }
+  \"aws\": [{
```

Comment:
Oh, I was thinking of something like:

```json
{
  "myCompany": {
    "provider": "aws",
   ...
  }, 
  "myPersonalProject": {
    "provider": "azure", 
   ... 
  } 
}
```

Reply:
So I did not go that route because this config should not really be application/project specific, but just for different cloud credentials / providers. My thought was there would be a one-to-many relationship between a cloud credential and an app/project. The application specific configuration as you describe it would come eventually, be checked into the repo and reference the cloud credential alias from this credentials configuration which really ought to be called credentials.json instead of deploy.json. Region and things like whether we use spot instances would eventually belong in that application specific configuration.

Reply:
So that's why I called the first one `"myCompany"` implying many projects under those credentials. The only difference here between your's and this one is pulling the alias out as the key in a hashmap instead of being another field within an array of JSON values?

Reply:
Ah I see, your point is just to key by the `cloudAlias` instead of arrays of cloud `provider`. That makes sense. I understood something else since I would probably use the same `cloudAlias` for all my personal projects.

Reply:
The other reason I had not done that is because the schema for `GCP` or `AZURE` might look very different from the ones `AWS` ones so it will be more complex or awkward to enforce a schema on this object, but it makes more sense to the user anyways